tests/integration/distribution.go

```diff
@@ -338,8 +340,156 @@ func (s *CCVTestSuite) TestEndBlockRD() {
 	}
 }
 
+// TestSendRewardsToProvider is effectively a unit test for TestSendRewardsToProvider(),
+// but is written as an integration test to avoid excessive mocking.
+func (s *CCVTestSuite) TestSendRewardsToProvider() {
+	testCases := []struct {
+		name           string
+		setup          func(sdk.Context, *consumerkeeper.Keeper, icstestingutils.TestBankKeeper)
+		expError       bool
+		tokenTransfers int
```

Comment:
Potentially this could be a `bool` because it would seem it only takes 2 values and then later on just check if `commitments` has one or zero elements?

Reply:
Sure, but like this we avoid an extra if condition.